pandas/io/html.py

```diff
@@ -734,13 +734,17 @@ def _parse(flavor, io, match, header, index_col, skiprows,
     ret = []
     for table in tables:
         try:
-            ret.append(_data_to_frame(data=table,
-                                      header=header,
-                                      index_col=index_col,
-                                      skiprows=skiprows,
-                                      parse_dates=parse_dates,
-                                      tupleize_cols=tupleize_cols,
-                                      thousands=thousands))
+            frame = _data_to_frame(data=table,
+                                   header=header,
```

Comment:
Yes, test_html.py:TestReadHtml.test_empty_tables expect a StopIteration to indicate empty tables.